README.md

```diff
@@ -172,6 +172,46 @@ TID is a 32-bit integer to identify threads created with `wasi_thread_spawn`.
     For example, it can be used to indicate the main thread, which doesn't
     have a TID in the current version of this proposal.
 
+### Process
+
+* A process is a group of threads.
+
+* The main thread starts with a process which only contains
+  the main thread.
+
+* Threads created by a thread in a process using `wasi_thread_spawn`
+  are added to the process.
+
+* When a thread is terminated, it's removed from the process.
+
+### Voluntary thread termination
+
+A thread can terminate itself voluntarily by returning from
+`wasi_thread_start`.
+
+### Changes to WASI `proc_exit`
+
+With this proposal, the `proc_exit` function takes extra responsibility
+to terminate all threads in the process, not only the calling one.
+
+Any of the threads in the process can call `proc_exit`.
+
+### Traps
+
+When a thread caused a trap, it terminates all threads in the process
+similarly to `proc_exit`.
+
+### Process exit status
+
+If one or more threads call WASI `proc_exit` or raise a trap,
+one of them is chosen by the runtime to represent the exit status
+of the process.
+It's non deterministic which one is chosen.
+
+If all the threads in the process have been terminated without calling
```

Comment:
How about, "If that last running thread in a process terminates without calling `proc_exit` or trapping it is treated as if ... "

Reply:
it's not only about the last running thread.

consider a process with two threads.

1. a thread terminated itself by calling `proc_exit(50)`
2. another thread, which is `the last running thread`, terminates itself by returning from `wasi_thread_spawn` before the runtime terminates it for `proc_exit`.
3. the exit code of the process should be 50.

i think your wording can be interpreted as the exit code of the process can be 0.

Reply:
My my understanding, in your example above `proc_exit(50)` causes all threads to be terminated, so its not possible (2) to occor.   

Another way of putting it "proc_exit` does not complete until all threads are terminated, so not thread can exist after `proc_exit` is complete.

Reply:
> My my understanding, in your example above `proc_exit(50)` causes all threads to be terminated, so its not possible (2) to occor.

if things work in a lockstep, maybe.
but we are talking about threads...

> Another way of putting it "proc_exit`does not complete until all threads are terminated, so not thread can exist after`proc_exit` is complete.

in that case, i guess you need to say the thread which called `proc_exit` is terminated after all other threads are terminated. i feel it's more complicated than the current sentence.

Reply:
I think it would odd if we allowed any threads to outlive the thread that calls `proc_exit`.. the whole point of that system call is to bring down all the threads so logically nothing else can happen on any thread one that syscall completes.

I don't want to hold up this PR, even though I don't love the way this sentence is phrased.   I guess we can land this as it stands and try to improve it later.